### PR TITLE
feat: 🎸 add parquet worker

### PR DESCRIPTION
**Pull request by severo**
missing:
- [x] unit test: download the parquet files and ensure the data are the same
- [x] docker compose
- [x] e2e
- [x] Helm
- [x] README, DEVELOPER GUIDE
- [x] fix an issue: we receive an "update" webhook every time we write to the `ref/convert/parquet` branch, which generates a new job -> updates are looping. It's OK to receive the webhook, but we should skip the job because the git version of `main` and the worker version should be the same as the cached entry. Fixed with https://github.com/huggingface/datasets-server/pull/651/commits/56614570e15a1c089286ed70d1f7cb5b0904cac7
- [x] docs
- [x] openapi
- [x] fix e2e: the sum of all the docker images is too large for the Github Actions storage, e2e action breaks.
some examples:
- [x] monitor the storage used for datasets cache, and possibly delete cache after some time. Internal chat: https://huggingface.slack.com/archives/C01TETY5V8S/p1669805539127299. For now, prod has access to 4TB, which gives sufficient space to start. We should delete the cache data for every dataset after a job.

First datasets with parquet files:
- `duorc`: [files](https://huggingface.co/datasets/duorc/tree/refs%2Fconvert%2Fparquet), [API](https://datasets-server.huggingface.co/parquet?dataset=duorc)
- `rotten_tomatoes`: [files](https://huggingface.co/datasets/rotten_tomatoes/tree/refs%2Fconvert%2Fparquet), [API](https://datasets-server.huggingface.co/parquet?dataset=rotten_tomatoes)
- `severo/mnist`: [files](https://huggingface.co/datasets/severo/mnist/tree/refs%2Fconvert%2Fparquet), [API](https://datasets-server.huggingface.co/parquet?dataset=severo/mnist)
- `severo/glue`: [files](https://huggingface.co/datasets/severo/glue/tree/refs%2Fconvert%2Fparquet), [API](https://datasets-server.huggingface.co/parquet?dataset=severo/glue)
- `severo/danish-wit`: [files](https://huggingface.co/datasets/severo/danish-wit/tree/refs%2Fconvert%2Fparquet), [API](https://datasets-server.huggingface.co/parquet?dataset=severo/danish-wit)

Issues:
- [ ] https://huggingface.co/datasets/Elite35P-Server/EliteVoiceProject gives a 500 error (it's a gated dataset, and we get a "forbidden" error). It should have been ignored because it's not on the list of supported datasets. Update: `Elite35P-Server/EliteVoiceProject` is no more gated, btw.
  Same with `SciSearch/wiki-data`, `TREC-AToMiC/AToMiC-Images-v0.1`, `TREC-AToMiC/AToMiC-Qrels-v0.1`, `TREC-AToMiC/AToMiC-Texts-v0.1`, `bigcode/the-stack` (it has extra fields), `bigcode/the-stack-dedup` (same), `mitclinicalml/clinical-ie`
- [ ] `severo/embellishments` and `severo/wit`: "An error occurred while generating the dataset"
- [ ] `severo/winogavil`: `Repo id must be in the form 'repo_name' or 'namespace/repo_name': 'datasets/nlphuji/winogavil'. Use `repo_type` argument if needed.`???

once deployed:
- [x] allow parquet conversion for all the datasets (see https://github.com/huggingface/datasets-server/blob/worker-parquet/chart/env/prod.yaml#L197)
- [x] increase the resources (number of nodes) in production
- [ ] rapid API: https://rapidapi.com/studio/api_0971eb45-d045-4337-be04-cec27c1122b0/client
- [ ] postman: https://www.postman.com/winter-station-148777/workspace/hugging-face-apis/collection/23242779-d068584e-96d1-4d92-a703-7cb12cbd8053

### CODE REVIEW

**Comment by severo** [comment]
The e2e tests are failing because the docker images are too big :(

> You are running out of disk space. The runner will stop working when the machine runs out of disk space. Free space left: 0 MB

**Comment by severo** [comment]
FYI the main cause for the size of the workers images are two dependencies: pytorch and tensorflow.

One possible solution is to use a shared base image for the three workers' docker images since the set of dependencies is very similar.

**Comment by severo** [comment]
the e2e tests should be able to run now because the three workers use the same docker image.

**Comment by severo** [comment]
Note sure why, but the documentation build is failing now... cc @mishig25 maybe

**Comment by severo** [comment]
After 12 hours, processing only the new and the updated datasets, we now get 70 datasets with valid parquet files in the `refs/convert/parquet` branch.

Here is the list:

<details><summary>deploy</summary>
<a href="https://datasets-server.huggingface.co/parquet?dataset=severo/glue">severo/glue</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=severo/mnist">severo/mnist</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=severo/danish-wit">severo/danish-wit</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=tbrugger/SBD-Multilingual-Italian">tbrugger/SBD-Multilingual-Italian</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=duorc">duorc</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=rotten_tomatoes">rotten_tomatoes</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=tbrugger/SBD-Multilingual-French">tbrugger/SBD-Multilingual-French</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=alexandrainst/danish-wit">alexandrainst/danish-wit</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=izou3/Food-Prototype">izou3/Food-Prototype</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=adrienheymans/imdb-movie-genres">adrienheymans/imdb-movie-genres</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=gngpostalsrvc/BERiT_vocalized">gngpostalsrvc/BERiT_vocalized</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=gngpostalsrvc/Tanakh">gngpostalsrvc/Tanakh</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=Whispering-GPT/whisper-transcripts-ml-street-talk">Whispering-GPT/whisper-transcripts-ml-street-talk</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=izou3/Test_Food">izou3/Test_Food</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=eminecg/petitions_29-ds">eminecg/petitions_29-ds</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=grosenthal/lat_en_loeb_whitaker_split">grosenthal/lat_en_loeb_whitaker_split</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=izou3/Food-Prototype-Bruce">izou3/Food-Prototype-Bruce</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=tbrugger/full_french">tbrugger/full_french</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=tbrugger/full_german">tbrugger/full_german</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=tbrugger/full_english">tbrugger/full_english</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=tbrugger/full_italian">tbrugger/full_italian</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=tbrugger/full_spanish">tbrugger/full_spanish</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=ravitejac/xeekfile5">ravitejac/xeekfile5</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=nbalepur/us_news_colleges_desc">nbalepur/us_news_colleges_desc</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=alighasemi/farsi_paraphrase_detection">alighasemi/farsi_paraphrase_detection</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=Zombely/pl-text-images">Zombely/pl-text-images</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=izou3/Food_Final">izou3/Food_Final</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=sms_spam">sms_spam</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=SatCat/github-issues">SatCat/github-issues</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=nbalepur/imdb_actor_bio">nbalepur/imdb_actor_bio</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=SatCat/github-embeddings-issues">SatCat/github-embeddings-issues</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=matallanas/test_audio">matallanas/test_audio</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=Thanapon1998/TH_corpora_parliament_processed">Thanapon1998/TH_corpora_parliament_processed</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=liyongsea/PTB-XL-ss">liyongsea/PTB-XL-ss</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=matallanas/test_transcript_mp3">matallanas/test_transcript_mp3</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=lucadiliello/trecqa">lucadiliello/trecqa</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=lucadiliello/wikiqa">lucadiliello/wikiqa</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=matallanas/test_transcript_aac">matallanas/test_transcript_aac</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/pip">open-source-metrics/pip</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=matallanas/test_transcript_alac">matallanas/test_transcript_alac</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/stars">open-source-metrics/stars</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/huggingface_hub-dependents">open-source-metrics/huggingface_hub-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/hub-docs-dependents">open-source-metrics/hub-docs-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/evaluate-dependents">open-source-metrics/evaluate-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/accelerate-dependents">open-source-metrics/accelerate-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=matallanas/test_transcript_m4a">matallanas/test_transcript_m4a</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/datasets-dependents">open-source-metrics/datasets-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/optimum-dependents">open-source-metrics/optimum-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=matallanas/test_transcript_opus">matallanas/test_transcript_opus</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=ScandEval/scandiqa-da-mini">ScandEval/scandiqa-da-mini</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=ScandEval/scandiqa-no-mini">ScandEval/scandiqa-no-mini</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=ScandEval/scandiqa-sv-mini">ScandEval/scandiqa-sv-mini</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/pytorch-image-models-dependents">open-source-metrics/pytorch-image-models-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/gradio-dependents">open-source-metrics/gradio-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/tokenizers-dependents">open-source-metrics/tokenizers-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/diffusers-dependents">open-source-metrics/diffusers-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/transformers-dependents">open-source-metrics/transformers-dependents</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/tabular-classification-checkpoint-downloads">open-source-metrics/tabular-classification-checkpoint-downloads</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/feature-extraction-checkpoint-downloads">open-source-metrics/feature-extraction-checkpoint-downloads</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/question-answering-checkpoint-downloads">open-source-metrics/question-answering-checkpoint-downloads</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/automatic-speech-recognition-checkpoint-downloads">open-source-metrics/automatic-speech-recognition-checkpoint-downloads</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/fill-mask-checkpoint-downloads">open-source-metrics/fill-mask-checkpoint-downloads</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=matallanas/test_transcript_vorbis">matallanas/test_transcript_vorbis</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/text-generation-checkpoint-downloads">open-source-metrics/text-generation-checkpoint-downloads</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=liamvbetts/sarcastic-news-headlines-1">liamvbetts/sarcastic-news-headlines-1</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/token-classification-checkpoint-downloads">open-source-metrics/token-classification-checkpoint-downloads</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/text2text-generation-checkpoint-downloads">open-source-metrics/text2text-generation-checkpoint-downloads</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=liamvbetts/sarcastic-news-headlines-v2">liamvbetts/sarcastic-news-headlines-v2</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=open-source-metrics/model-repos-stats">open-source-metrics/model-repos-stats</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=SpeedOfMagic/xsum_tiny_ood">SpeedOfMagic/xsum_tiny_ood</a>
<a href="https://datasets-server.huggingface.co/parquet?dataset=kidd2012/kidd-github-issues">kidd2012/kidd-github-issues</a>
</details>

**Comment by severo** [comment]
Mergggggin
